object_template.go

```diff
@@ -0,0 +1,86 @@
+package v8go
+
+// #include <stdlib.h>
+// #include "v8go.h"
+import "C"
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"runtime"
+	"unsafe"
+)
+
+// PropertyAttribute are the attributes for a property
+type PropertyAttribute uint8
+
+const (
+	// None.
+	None PropertyAttribute = 0
+	// ReadOnly, ie. not writable.
+	ReadOnly = 1 << iota
+	// DontEnum, ie. not enumerable.
+	DontEnum
+	// DontDelete, ie. not configurable.
+	DontDelete
+)
+
+// ObjectTemplate is used to create objects at runtime.
+// Properties added to an ObjectTemplate are added to each object created from the ObjectTemplate.
+type ObjectTemplate struct {
+	ptr C.ObjectTemplatePtr
+	iso *Isolate
+}
+
+// NewObjectTemplate creates a new ObjectTemplate.
+// The *ObjectTemplate can be used as a v8go.ContextOption to create a global object in a Context.
+func NewObjectTemplate(iso *Isolate) (*ObjectTemplate, error) {
+	if iso == nil {
+		return nil, errors.New("object_template: failed to create new ObjectTemplate: isolate cannot be <nil>")
```

Comment:
kind of odd to prefix with object_template here -- consider omitting or using v8go. You could also just simpify dramatically via:
"isolate cannot be nil"

Reply:
Fair enough. Should really only prefix with the the package name ie. `v8go` but I was following a previous pattern (that I wrongly did for the value struct). Will amend all error messages for consistency.